### PR TITLE
[agent] Add support for histograms with control plane metrics

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add support for histograms with control plane metrics behind a feature gate.

### CODE REVIEW

File: examples/controlplane-histogram-metrics/rendered_manifests/configmap-agent.yaml [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+            scrape_interval: 10s
+            static_configs:
+            - targets:
+              - ${K8S_POD_IP}:8889
+      receiver_creator:
+        receivers:
+          prometheus/coredns:
+            config:
+              extraDimensions:
+                metric_source: k8s-coredns
```
Comment (dmitryax) [inline examples/controlplane-histogram-metrics/rendered_manifests/configmap-agent.yaml:192]: Where is this coming from? I don't think prometheus receiver has `extraDimensions` config option

Comment (jvoravong) [reply examples/controlplane-histogram-metrics/rendered_manifests/configmap-agent.yaml:192]: +1
For simplicity, I might consider coordinating with the team responsible for creating default customer content to eliminate the use of "metric_source: *" attributes/dimensions for K8s control plane content.
- Breaking Change: This change will require a "breaking change" entry in the changelog due to the removal of attributes.
- Rationale: Our metric names and metadata are already unique, making "metric_source: *" attributes redundant for K8s control plane content.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 2]
```diff
@@ -884,6 +1016,20 @@ service:
         {{- end }}
         {{- end }}
     {{- end }}
+
+    {{- if and (eq (include "splunk-otel-collector.splunkO11yEnabled" .) "true") .Values.featureGates.useControlPlaneMetricsHistogramData }}
+    metrics/histograms:
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:1021]: I haven't noticed any filtering for the non-histogram metrics collected by the new Prometheus receivers. Are we using a default set of metrics or applying a filter somewhere? Without filtering, we might duplicate existing control plane metrics and inadvertently collect X number of new or custom control plane metrics.

File: examples/controlplane-histogram-metrics/rendered_manifests/configmap-agent.yaml (hunk 1 above)
Comment (atoulme) [reply examples/controlplane-histogram-metrics/rendered_manifests/configmap-agent.yaml:192]: Sorry, more work needed.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl (hunk 2 above)
Comment (atoulme) [reply helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:1033]: We should activate one or the other if you scroll up ; we will need to test for it.

**Comment by jvoravong** [comment]
We also probably have to tackle updating the the not-openshift rules for the receivers to values like these. This way we include rules for older and current labels of control plane pods, testing might be blocked if the new labels are not added.
 ```
       rule: type == "pod" && (labels["k8s-app"] == "kube-dns" ||  || (labels["tier"] == "control-plane" && labels["component"] == "kube-dns"))
        rule: type == "pod" && (labels["k8s-app"] == "etcd-manager-events" || labels["k8s-app"] == "etcd-manager-main" || (labels["tier"] == "control-plane" && labels["component"] == "etcd"))
        rule: type == "pod" && (labels["k8s-app"] == "kube-controller-manager" || (labels["tier"] == "control-plane" && labels["component"] == "kube-controller-manager"))
        rule: type == "port" && port == 443 && (labels["k8s-app"] == "kube-apiserver" || (labels["tier"] == "control-plane" && labels["component"] == "kube-apiserver"))
        rule: type == "pod" && (labels["k8s-app"] == "kube-proxy" || (labels["tier"] == "control-plane" && labels["component"] == "kube-proxy"))
        rule: type == "pod" && (labels["k8s-app"] == "kube-scheduler" || (labels["tier"] == "control-plane" && labels["component"] == "kube-scheduler"))
```

File: .chloggen/controlplanemetricshistogram.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+component: agent
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add a feature gate `useControlPlaneMetricsHistogramData`
+# One or more tracking issues related to the change
+issues: [1372]
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  This feature gate allows to gather control plane metrics and send them as histogram data to Observability Cloud.
```
Comment (jvoravong) [inline .chloggen/controlplanemetricshistogram.yaml:13]:
```suggestion
This feature gate enables the collection of control plane metrics as before, and now adds the ability to send histogram data to Observability Cloud. 
It also replaces Kubernetes control plane Smart Agent receivers with OpenTelemetry Prometheus receivers.
```

**Pull request merged**